src/moonlink/src/storage/iceberg/iceberg_table_manager.rs

```diff
@@ -277,6 +278,26 @@ impl IcebergTableManager {
         Ok(())
     }
 
+    /// Util function to get the data file to file indice mapping.
+    /// NOTICE: it performs a sequential scan on all file indices, which assumes file indices number to be acceptable, otherwise we need to construct the mapping from manifest entries.
+    fn get_data_file_to_file_indice_mapping(
+        persisted_file_indices: &[MooncakeFileIndex],
+    ) -> HashMap<MooncakeDataFileRef, MooncakeFileIndex> {
+        let data_file_num = persisted_file_indices
+            .iter()
+            .map(|cur_file_indice| cur_file_indice.files.len())
+            .sum();
+        let mut data_file_to_file_indice = HashMap::with_capacity(data_file_num);
+        for cur_file_indice in persisted_file_indices.iter() {
+            for cur_data_file in cur_file_indice.files.iter() {
+                let old_entry =
+                    data_file_to_file_indice.insert(cur_data_file.clone(), cur_file_indice.clone());
+                assert!(old_entry.is_none());
```

Comment:
The assertion serves partially as the test, which fails if multiple data files get assigned the same file id.